lib/document/stateTransition/applyDocumentsBatchTransitionFactory.js

```diff
@@ -0,0 +1,65 @@
+const AbstractDocumentTransition = require(
+  './documentTransition/AbstractDocumentTransition',
+);
+
+/**
+ * @param {StateRepository} stateRepository
+ * @param {applyDocumentsBatchTransitionToModels} applyDocumentsBatchTransitionToModels
+ *
+ * @returns {applyDocumentsBatchTransition}
+ */
+function applyDocumentsBatchTransitionFactory(
+  stateRepository,
+  applyDocumentsBatchTransitionToModels,
+) {
+  /**
+   * Apply documents batch state transition
+   *
+   * @typedef applyDocumentsBatchTransition
+   *
+   * @param {DocumentsBatchTransition} stateTransition
+   *
+   * @return {Promise<void>}
+   */
+  async function applyDocumentsBatchTransition(stateTransition) {
+    const existingDocuments = stateTransition.getTransitions()
+      .reduce(async (result, documentTransition) => {
+        const documentId = documentTransition.getId();
+
+        const [existingDocument] = await stateRepository.fetchDocuments(
+          stateTransition.getDataContractId(),
+          documentTransition.getType(),
+          { where: [['$id', '==', documentId]] },
+        );
+
+        return {
+          ...result,
+          [documentId]: existingDocument,
+        };
+      }, {});
+
+    const documents = await applyDocumentsBatchTransitionToModels(
```

Comment:
do we still need this?

Reply:
@shumkov Merge would result in a huge function, do we want that?

Reply:
In this case, we don't need to prepare/iterate documents map so it will be simpler.